### PR TITLE
Silas - Make matching of emails case insensitive

### PR DESCRIPTION
**Pull request by said-moj**
## What does this pull request do?

Make matching of emails case insensitive.

## Any other changes that would benefit highlighting?

Intentionally left blank.

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

**Review by joselopez-moj — DISMISSED** [review]

File: cla_backend/apps/cla_auth/backend.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 class EntraAccessTokenAuthenticationBackend(ModelBackend):
     def authenticate(self, entra_id_email=None):
         try:
-            user = User.objects.get(email=entra_id_email, is_active=True)
+            user = User.objects.get(email__iexact=entra_id_email, is_active=True)
```
Comment (skdejong) [inline cla_backend/apps/cla_auth/backend.py]: This looks good and the test covers the intended behaviour. One possible edge case: if duplicate active users exist with emails differing only by case, `email__iexact` could raise `MultipleObjectsReturned` and authentication would fail silently due to the broad `except Exception`. Are emails guaranteed unique case-insensitively elsewhere?

Comment (said-moj) [reply cla_backend/apps/cla_auth/backend.py]: @skdejong I think the edge case of `MultipleObjectsReturned` silently failing is a valid edge case that we need to consider. However if two users have the same email address but in different casing then we consider those two having the same email address.

**Review by skdejong — DISMISSED** [review]

**Review by joselopez-moj — DISMISSED** [review]

File: cla_backend/apps/cla_auth/authentication.py [hunk 2]
```diff
@@ -171,6 +171,9 @@ def get_or_create_user(self, payload):
 
         try:
             user = authenticate(entra_id_email=email)
+        except User.MultipleObjectsReturned:
+            logger.error("Multiple users found with the email address  %s", email, exc_info=True)
```
Comment (skdejong) [inline cla_backend/apps/cla_auth/authentication.py]: Is it worth changing this line to 

```
logger.exception(
    "Multiple users found with the email address %s",
    email
)
```

in order to satisfy Sonar and make the intent clearer?

Comment (said-moj) [reply cla_backend/apps/cla_auth/authentication.py]: @skdejong this has been added now.

btw logger.exception calls logger.error with `exc_info=True`

**Review by skdejong — APPROVED** [review]

**Review by joselopez-moj — APPROVED** [review]
